src/Internal/ProductAttributesLookup/DataRegenerator.php

```diff
@@ -367,64 +368,24 @@ private function enable_or_disable_lookup_table_usage( $enable ) {
 	}
 
 	/**
-	 * Add a metabox in the product page with a button to regenerate the product attributes lookup data for the product.
-	 */
-	private function add_product_regeneration_metabox() {
-		if ( ! $this->can_do_per_product_regeneration() ) {
-			return;
-		}
-
-		add_meta_box(
-			'woocommerce-product-foobars',
-			__( 'Lookup data', 'woocommerce' ),
-			function() {
-				$this->metabox_output();
-			},
-			'product',
-			'side',
-			'low'
-		);
-	}
-
-	/**
-	 * HTML output for the lookup data regeneration metabox.
-	 */
-	private function metabox_output() {
-		// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
-		wp_nonce_field( 'regenerate-attributes-lookup-data', '_wc_regenerate_attributes_lookup_data_nonce' );
-		?>
-		<p><?php echo __( 'Click to regenerate the product attributes lookup data for this product:', 'woocommerce' ); ?></p>
-		<button class="button button-primary button-large" name="woocommerce-product-lookup-action" value="regenerate-attributes-lookup-data">
-		<?php echo __( 'Regenerate', 'woocommerce' ); ?>
-		</button>
-		<?php
-		// phpcs:enable WordPress.Security.EscapeOutput.OutputNotEscaped
-	}
-
-	/**
-	 * Hook on the 'save_post_product' filter to regenerate the product attributes lookup data when the regenerate metabox button was pressed.
+	 * Check if everything is good to go to perform a per product lookup table data regeneration
+	 * and throw an exception if not.
 	 *
-	 * @param int $product_id The product id.
+	 * @param mixed $product_id The product id to check the regeneration viability for, or null to skip product check.
+	 * @throws \Exception Something prevents the regeneration from starting.
 	 */
-	private function on_save_product( int $product_id ) {
-		if (
-			! wp_verify_nonce( ArrayUtil::get_value_or_default( $_POST, '_wc_regenerate_attributes_lookup_data_nonce' ), 'regenerate-attributes-lookup-data' ) ||
-			! $this->can_do_per_product_regeneration() ||
-			'regenerate-attributes-lookup-data' !== ArrayUtil::get_value_or_default( $_POST, 'woocommerce-product-lookup-action' ) ||
-			! wc_get_product( $product_id )
-		) {
-			return;
+	private function check_can_do_lookup_table_regeneration( $product_id = null ) {
```

Comment:
Small nitpick here but we probably can remove the `check` from the function name because `can` already means to check.